### PR TITLE
vscode-extensions/update-8

### PR DESCRIPTION
**Pull request by eadwu**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vscode-extensions/python/default.nix [hunk 1]
```diff
@@ -37,19 +38,26 @@ let
       sha256 = languageServerSha256;
     };
   };
+
+  libcMusl = runCommand "libc-musl-x86_64" {} ''
+    mkdir -p $out/lib
+    cp -pdv ${musl}/lib/libc.so $out/lib/libc.musl-x86_64.so.1
+    ln -s $out/lib/libc.musl-x86_64.so.1 $out/lib/libc.musl-x86_64.so
```
Comment (eadwu) [inline pkgs/misc/vscode-extensions/python/default.nix:45]: Should this go into the `musl` derivation?

Comment (jonringer) [reply pkgs/misc/vscode-extensions/python/default.nix:45]: most likely, but that can be a separate PR

**Review by jonringer — COMMENTED** [review]
having issues with the python extension:

```
trying https://ms-python.gallery.vsassets.io/_apis/public/gallery/publisher/ms-python/extension/python/2020.5.80290/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 11 44.1M   11 5156k    0     0   136k      0  0:05:31  0:00:37  0:04:54  569k
curl: (18) transfer closed with 40970331 bytes remaining to read
** Resuming transfer from byte position 5280295
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 44.1M    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
```

**Comment by eadwu** [comment]
Depends on https://github.com/NixOS/nixpkgs/pull/89112 now.

I think the `src` might be the server's problem (sounds really familiar but not sure), tried fetching the vsix from Github?

File: pkgs/misc/vscode-extensions/python/default.nix (hunk 1 above)
Comment (eadwu) [reply pkgs/misc/vscode-extensions/python/default.nix]: https://github.com/NixOS/nixpkgs/pull/89112

**Comment by jonringer** [comment]
> I think the `src` might be the server's problem (sounds really familiar but not sure), tried fetching the vsix from Github?

probably, not sure how to clear my local instance :(

**Comment by eadwu** [comment]
Does the Github VSIX have any problems in regards to fetching the zip? The extension itself seems to working fine from what I've used.

**Comment by eadwu** [comment]
From a raw comparison using `diff` and `find`, it looks like `python-language-server` has all the files, just in a different directory format, I'll see if I can use that for the extension.

**Comment by jonringer** [comment]
thanks for continuing to work on this, but do you mind squashing the intermediate bumps? Should only be the 2 extensions

**Comment by eadwu** [comment]
Squashed.

**Review by jonringer — APPROVED** [review]

**Pull request merged**